src/components/hooks/use-platform-query-param.tsx

```diff
@@ -1,18 +1,28 @@
 import { useEffect, useState } from 'react'
-import { useQueryParam, StringParam } from 'use-query-params'
+import queryString from 'query-string'
+
+type MyQuery = {
+    platform: string
+}
+
+const qparam =
+    typeof window !== 'undefined' &&
+    location.search &&
+    (queryString.parse(location.search) as MyQuery)
```

Comment:
```suggestion
    (queryString.parse(location.search))
```
is this casting required? Maybe you can define the return type using `ReturnType<T>` https://www.typescriptlang.org/docs/handbook/utility-types.html